Windows/XinputDevice.cpp

```diff
@@ -338,3 +342,22 @@ void XinputDevice::ApplyButtons(int pad, const XINPUT_STATE &state) {
 		}
 	}
 }
+
+
+void XinputDevice::ApplyVibration(int pad) {
+	if (PSP_IsInited) {
+		XINPUT_VIBRATION vibration;
+		ZeroMemory(&vibration, sizeof(XINPUT_VIBRATION));
+		vibration.wLeftMotorSpeed = 0;
+		vibration.wRightMotorSpeed = 0;
+		
+		engine_ = new CWCheatEngine(gameID_);
+		if (engine_) {
+			vibration.wLeftMotorSpeed = engine_->GetLefttVibration(); // use any value between 0-65535 here
+			vibration.wRightMotorSpeed = engine_->GetRightVibration(); // use any value between 0-65535 here
+		}
+		WARN_LOG(COMMON, "PAD TEST LEFT: %d, RIGHT: %d, PAD: %d", vibration.wLeftMotorSpeed, vibration.wRightMotorSpeed, pad);
+		XInputSetState(pad, &vibration);
```

Comment:
I think it might be worth not calling this if the vibration hasn't changed since the last frame. But not sure.

Reply:
Instead of doing this, the Right thing would be to do almost exactly what we do with `XInputGetState`.  If you search the file for XInputGetState, you can basically double up most of the lines and make it Set instead of Get.  Then you don't need the xinput.lib.

-[Unknown]